guides/source/generators.md

```diff
@@ -475,35 +496,255 @@ after_bundle do
+
+```ruby
+file "app/components/foo.rb", <<-CODE
+  class Foo
+  end
+CODE
+```
+
+The above will create the `app/components` directory and put `foo.rb` in there.
+
+### rakefile
+
+The `rakefile` method creates a new rake file under `lib/tasks` with the given tasks:
+
+```ruby
+rakefile("bootstrap.rake") do
+  <<-TASK
+    namespace :boot do
+      task :strap do
+        puts "i like boots!"
+      end
+    end
+  TASK
+end
+```
+
+The above creates `lib/tasks/bootstrap.rake` with a `boot:strap` rake task.
+
+### generate
+
+You can even call a generator from inside a `template.rb`. The following runs the `scaffold` rails generator with the given arguments:
+
+```ruby
+generate(:scaffold, "person", "name:string", "address:text", "age:number")
+```
+
+### run
+
+This method executes an arbitrary command. Let's say you want to remove the `README.rdoc` file:
+
+```ruby
+run "rm README.rdoc"
+```
+
+### rails_command
+
+You can run the Rails commands in the generated application. Let's say you want to migrate the database at some point in the template ruby script:
+
+```ruby
+rails_command "db:migrate"
+```
+
+You can also run commands with a different Rails environment:
+
+```ruby
+rails_command "db:migrate", env: "production"
+```
+
+You can also run commands that should abort application generation if they fail:
+
+```ruby
+rails_command "db:migrate", abort_on_failure: true
+```
+
+### route
+
+This method adds an entry to the `config/routes.rb` file. To make `PeopleController#index` the default page for the application, we can add:
+
+```ruby
+route "root to: 'person#index'"
+```
+
+### inside
+
+This mehtod enables you to run a command from a given directory. For example, if you have a copy of edge rails that you wish to symlink from your new apps, you can do this:
+
+```ruby
+inside("vendor") do
+  run "ln -s ~/commit-rails/rails rails"
+end
+```
+
+### ask
+
+The `ask()` methods allows you to get feedback from the user and use it in your templates. Let's say you want your user to name the new shiny library you're adding:
+
+```ruby
+lib_name = ask("What do you want to call the shiny library ?")
+lib_name << ".rb" unless lib_name.index(".rb")
+
+lib lib_name, <<-CODE
+  class Shiny
+  end
+CODE
+```
+
+### yes? or no?
+
+These methods let you ask questions from templates and decide the flow based on the user's answer. Let's say you want to prompt the user to run migrations:
+
+```ruby
+rails_command("db:migrate") if yes?("Run database migrations?")
+# no? questions acts the opposite of yes?
+```
+
+### git
+
+Rails templates let you run any git command:
+
+```ruby
+git :init
+git add: "."
+git commit: "-a -m 'Initial commit'"
+```
+
+### after_bundle
+
+The method registers a callback to be executed after the gems are bundled. For example, it would make sense to run the "install" command for gems `tailwindcss-rails` and `devise` only after those gems are bundled:
```

Comment:
```suggestion
This method registers a callback to be executed after the gems are bundled. For example, it would make sense to run the "install" command for `tailwindcss-rails` and `devise` only after those gems are bundled:
```